HTTP_API_Guide.md

```diff
@@ -0,0 +1,206 @@
+HTTP API
+========
+
+The [Bloomberg Open API] provides access to market data. The
+[Bloomberg HTTP API] makes the [Open API] available via HTTP: clients POST
+commands to an HTTP server, which will respond with the relevant data.
+
+The [HTTP API] is currently in development, and does not yet provide access
+to all features of the [Open API]. At this time it supports the Reference Data
+Service (i.e. /blp/refdata) and the API Field Service (/blp/apiflds).
+
+This document provides an overview of the [HTTP API]: the URLs to use and the
+formatting for the HTTP requests and responses. We assume, for the purposes of
+this doc, that the server at http://blpapihost.example.com/ is running an
+implementation of this API.
+
+This doc also includes basic info about some of the underlying [Open API]
+operations. For more details on the [Open API], refer to the
+[BLPAPI Developer's Guide]. This document will often refer to specific sections
+of the [Developer's Guide] where appropriate.
+
+[Bloomberg Open API]: http://bloomberglabs.com/api
+[Open API]: http://bloomberglabs.com/api
+[Bloomberg HTTP API]: http://github.com/bloomberg/blpapi-http
+[HTTP API]: http://github.com/bloomberg/blpapi-http
+[BLPAPI Developer's Guide]: http://www.bloomberglabs.com/files/2014/07/blpapi-developers-guide-2.54.pdf
+[Developer's Guide]: http://www.bloomberglabs.com/files/2014/07/blpapi-developers-guide-2.54.pdf
+
+
+Requesting data
+---------------
+
+To request data, a client makes an HTTP POST request on a URL of the following
+form:
+
+```
+https://blpapihost.example.com/request/blp/refdata/<operation>
+```
+
+Let's examine each segment of the path:
+
+* /request -- The Open API paradigm we're using, namely Request/Response.
+* /blp/refdata -- The service on which we're making the request. Currently /blp/refdata and /blp/apiflds are supported; more will be added in future versions.
+* &lt;operation&gt; -- The operation for this request. For /blp/refdata, the operations are listed in the Developer's Guide in section A.2.1 (and detailed in 7.2). Note that the URL omits the "Request" in the operation name, e.g. for the HistoricalDataRequest operation, use /HistoricalData
+
+The body of the POST should consist of a JSON object containing the parameters
+of the request. Each operation defines what parameters it uses; sections 7.2
+and A.2 of the [Developer's Guide] provide full details.
+
+As an example, the following will request the open and last price for IBM and
+Apple for each day during the period Jan 1 - 5, 2012.
+
+```
+curl -X POST https://blpapihost.example.com/request/blp/refdata/HistoricalData --data @- <<EOF
+{ "securities": ["IBM US Equity", "AAPL US Equity"],
+  "fields": ["PX_LAST", "OPEN"],
+  "startDate": "20120101",
+  "endDate": "20120105",
+  "periodicitySelection": "DAILY" }
+EOF
+```
+
+
+HTTP request headers
+--------------------
+
+The client may specify the following HTTP request headers to tell the server
+what sorts of responses it can use:
```

Comment:
It may be a good idea to partition which headers are required and optional. (i.e., Content-Type:) is more or less required.  @ericlu88 can you confirm?